src/System.ComponentModel.Primitives/src/System/ComponentModel/Component.cs

```diff
@@ -20,7 +20,7 @@ public class Component : MarshalByRefObject, IComponent
         private ISite _site;
         private EventHandlerList _events;
 
-        ~Component() => Dispose(false);
+        ~Component() { Dispose(false); }
```

Comment:
Should we rather disable the broken analyzer rule globally instead of working it around everywhere?

Reply:
My thinking is that the analyzer still provides value to other places. Imagine some new code gets ported, or written from scratch, and is using an actual empty finalizer. The analyzer rule would still catch these cases.

There is no functional difference between using expression bodies and statement bodies, and it took ~10 min to find and replace these. So I didn't thinking disabling it completely was worth the value.

Thoughts?

Reply:
> Imagine some new code gets ported, or written from scratch, and is using an actual empty finalizer

The chances that such code is introduced and not caught during codereview is very close to zero.

Reply:
I'm not sure there is a good way to disable just this one analyzer globally so lets just move forward with @eerhardt's current approach to get this PR unblocked and use issue https://github.com/dotnet/corefx/issues/32162 to find another solution for the analyzer if necessary.

Reply:
> I'm not sure there is a good way to disable just this one analyzer globally 

`<NoWarn>$(NoWarn);1821</NoWarn>` should do it.

Reply:
Nice I didn't realize it would be that simple. I will give that a go.

Reply:
NoWarn didn't work but can disable the rule in the ruleset file see https://github.com/dotnet/corefx/pull/32177.